src/MessageHandler/RedeemInvitationHandler.php

```diff
@@ -7,32 +7,38 @@
 use App\Message\RedeemInvitation;
 use App\Registration\InvitationProvider;
 use App\Registration\InvitationRedeemer;
+use App\Registration\UserProvider;
 use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
 use Symfony\Component\Messenger\MessageBusInterface;
 
 class RedeemInvitationHandler implements MessageHandlerInterface
 {
     private $invitationProvider;
+    private $userProvider;
     private $invitationRedeemer;
     private $messageBus;
 
     public function __construct(
         InvitationProvider $invitationProvider,
+        UserProvider $userProvider,
         InvitationRedeemer $invitationRedeemer,
         MessageBusInterface $messageBus
     ) {
         $this->invitationProvider = $invitationProvider;
+        $this->userProvider = $userProvider;
         $this->invitationRedeemer = $invitationRedeemer;
         $this->messageBus = $messageBus;
     }
 
     public function __invoke(RedeemInvitation $redeemInvitationMessage): void
     {
         $invitation = $this->invitationProvider->getOpenInvitation($redeemInvitationMessage->getInviteCode());
-        $this->invitationRedeemer->redeem($invitation, $redeemInvitationMessage->getInvitedUser());
+        $inviteduser = $this->userProvider->getUserByEmail($redeemInvitationMessage->getInvitedUser());
 
-        $this->messageBus->dispatch(new CreateInvitations($redeemInvitationMessage->getInvitedUser()));
+        $this->invitationRedeemer->redeem($invitation, $inviteduser);
 
-        $this->messageBus->dispatch(new InvitationRedeemed($invitation));
+        $this->messageBus->dispatch(new CreateInvitations((string) $inviteduser));
```

Comment:
I am not convinced that coupling the handler to a specific implementation of `__toString()` is a good idea. I think we should rather fetch the data we are interested in explicitly instead.